src/metabase/lib/remove_replace.cljc

```diff
@@ -424,3 +430,40 @@
                                                            new-join
                                                            %)
                                                         joins))))))
+
+(defn- specifies-default-fields? [query stage-number]
+  (let [fields (:fields (lib.util/query-stage query stage-number))]
+    (and fields
+         ;; Quick first check: if there are any implicitly-joined fields, it's not the default list.
+         (not (some (comp :source-field lib.options/options) fields))
+         (lib.equality/matching-column-sets? query stage-number fields
+                                             (lib.metadata.calculation/default-columns-for-stage query stage-number)))))
+
+(defn- normalize-fields-for-join [query stage-number join]
+  (if (#{:none :all} (:fields join))
+    ;; Nothing to do if it's already a keyword.
+    join
+    (cond-> join
+      (lib.equality/matching-column-sets?
+        query stage-number (:fields join)
+        (lib.metadata.calculation/returned-columns query stage-number (assoc join :fields :all)))
+      (assoc :fields :all))))
```

Comment:
The doc below promises normalizing the empty list to `:none`, but that doesn't seem to be happening.

Reply:
`:none` is actually the default, so having it removed is correct and minimal. I've fixed the doc.

Reply:
Will/should `:none` be removed during normalization? (In other words, are we really normalizing?)